### PR TITLE
[bitnami/mongodb-sharded] add podManagementPolicy config for statefulsets to enable parallel startup

### PR DESCRIPTION
**Pull request by oldluke92**
**Description of the change**

- making the statefulsets configurable to enable parallel startup of db pods
- Default to "OrderedReady" to stay backwards compatible to deployed clusters

**Benefits**
- this speeds up db startup and improves resilency against concecutive failures. For example if you got a setup using 3 data replicas and the "middle" one fails because of a synchronisation error, you will not be able to restart/move the 3rd one because the middle one not being ready prevents the third one from being created again.

**Possible drawbacks**

You tell me. Maybe there was a reason for not allowing parallel startup.

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: bitnami/mongodb-sharded/templates/config-server/config-server-statefulset.yaml [hunk 1]
```diff
@@ -11,6 +11,7 @@ spec:
       app.kubernetes.io/component: configsvr
   serviceName: {{ include "mongodb-sharded.fullname" . }}-headless
   replicas: {{ .Values.configsvr.replicas }}
+  podManagementPolicy: {{ $.Values.configsvr.podManagementPolicy }}
```
Comment (juan131) [inline bitnami/mongodb-sharded/templates/config-server/config-server-statefulset.yaml:14]: You don't need to use the `$.`. Suggestion:
```
  podManagementPolicy: {{ .Values.configsvr.podManagementPolicy }}
```

File: bitnami/mongodb-sharded/templates/shard/shard-arbiter-statefulset.yaml [hunk 2]
```diff
@@ -11,6 +11,7 @@ spec:
   selector:
     matchLabels: {{- include "mongodb-sharded.matchLabels" $ | nindent 6 }}
       app.kubernetes.io/component: shardsvr-arbiter
+  podManagementPolicy: {{ $.Values.shardsvr.arbiter.podManagementPolicy }}
```
Comment (juan131) [inline bitnami/mongodb-sharded/templates/shard/shard-arbiter-statefulset.yaml:14]: You don't need to use the `$.`. Suggestion:
```
  podManagementPolicy: {{ .Values.shardsvr.arbiter.podManagementPolicy }}
```

File: bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml [hunk 3]
```diff
@@ -11,6 +11,7 @@ spec:
   selector:
     matchLabels: {{- include "mongodb-sharded.matchLabels" $ | nindent 6 }}
       app.kubernetes.io/component: shardsvr
+  podManagementPolicy: {{ $.Values.shardsvr.dataNode.podManagementPolicy }}
```
Comment (juan131) [inline bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml:14]: You don't need to use the `$.`. Suggestion:
```
  podManagementPolicy: {{ .Values.shardsvr.dataNode.podManagementPolicy }}
```

**Comment by oldluke92** [comment]
Well it seems it doesn't work like that :(

**Comment by juan131** [comment]
You're right
Please revert the suggestions (except in the **config-server-statefulset.yaml** where it actually works)

**Comment by juan131** [comment]
Thanks so much!

**Comment by oldluke92** [comment]
It seems that the bootstrapping scripts for the data replicasets do not support a parallel initial spinup of the cluster nodes. This is a problem, as the problem described in the "benefits" section of this PR might arise (and did, in our deployment ..) when unsing OrderedReady. Shall I open an issue for that? I don't really have the time to search for that (and I would also not know where to start) ....
Unfortunately, the cluster can also not be deployed using OrderedReady and then changed to parallel later - because this is a field that cannot be changed....

**Pull request merged**